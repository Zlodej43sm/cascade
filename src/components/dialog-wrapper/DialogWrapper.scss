@use '../../styles/colors';
@use '../../styles/typography';
@use '../../styles/mixins';
@use '../../styles/sizes';

.container {
  @include mixins.themeBoxShadow(colors.$base-darkest);
  background-color: colors.$base-lightest;
  border-radius: 20px;
}

.header {
  padding: sizes.$base-3 sizes.$base-4;
  position: relative;
}

.title {
  @include typography.title-lg;
  padding-right: sizes.$base-4;
  color: colors.$cascade-secondary;
}

.closeButton {
  position: absolute;
  top: sizes.$base-3;
  right: sizes.$base-3;
}

.content {
  overflow: auto;
  padding: 0 sizes.$base-4 sizes.$base-4;
}
