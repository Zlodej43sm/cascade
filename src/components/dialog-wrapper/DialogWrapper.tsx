import React from 'react';

import CloseButton from 'lib/components/buttons/close-button/CloseButton';
import styles from './DialogWrapper.scss';

export type DialogWrapperProps = {
  title?: string
  children?: React.ReactNode
  width?: number | string
  height?: number | string
  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void
};

function DialogWrapper({ title, children, width, height, onClose }: DialogWrapperProps) {
  return (
    <section className={styles.container} style={{ width, height }} data-testid="dialog-wrapper-container">
      <header className={styles.header}>
        {title && (
          <div className={styles.title} data-testid="dialog-wrapper-title">
            {title}
          </div>
        )}
        <div className={styles.closeButton}>
          <CloseButton onClose={onClose} />
        </div>
      </header>
      <div className={styles.content}>
        {children}
      </div>
    </section>
  );
}

export default DialogWrapper;
