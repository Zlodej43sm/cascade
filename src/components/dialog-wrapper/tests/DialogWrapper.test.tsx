import React from 'react';
import { render } from '@testing-library/react';

import { dialogWrapperArguments } from '../__mocks__/DialogWrapper.mock';
import DialogWrapper from '../DialogWrapper';

describe('CardWrapper', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<DialogWrapper {...dialogWrapperArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
