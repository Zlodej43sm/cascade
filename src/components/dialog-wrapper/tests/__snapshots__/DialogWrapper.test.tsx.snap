// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper Should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="dialog-wrapper-container"
    style="width: 50%; height: 50%;"
  >
    <header
      class="header"
    >
      <div
        class="title"
        data-testid="dialog-wrapper-title"
      >
        Dialog title
      </div>
      <div
        class="closeButton"
      >
        <button
          aria-label="Close"
          class="button"
          data-testid="close-button"
          type="button"
        />
      </div>
    </header>
    <div
      class="content"
    >
      Child content
    </div>
  </section>
</DocumentFragment>
`;
